flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskManagerServices.java

```diff
@@ -529,11 +529,12 @@ public static ResourceProfile computeSlotResourceProfile(int numOfSlots, long ma
 		int managedMemoryPerSlotMB = (int) bytesToMegabytes(managedMemorySize / numOfSlots);
 		return new ResourceProfile(
 			Double.MAX_VALUE,
-			Integer.MAX_VALUE,
-			Integer.MAX_VALUE,
-			Integer.MAX_VALUE,
-			Integer.MAX_VALUE,
-			managedMemoryPerSlotMB,
+			MemorySize.MAX_VALUE,
+			MemorySize.MAX_VALUE,
+			// TODO: before operators separate on-heap/off-heap managed memory, we use on-heap managed memory to denote total managed memory
+			MemorySize.parse(managedMemoryPerSlotMB + "m"),
```

Comment:
This should basically be adjusted with the #9801, right?

Reply:
Yes